kairon/api/app/routers/bot/data.py

```
@@ -170,6 +171,25 @@ async def delete_cognition_data(
+async def delete_multiple_cognition_data(
+        row_ids: list[str] = Query(...),
+        current_user: User = Security(Authentication.get_current_user_and_bot, scopes=DESIGNER_ACCESS),
+):
+    """
+    Deletes multiple cognition content entries of the bot in bulk
+    """
+    try:
+        query = {"id__in": row_ids}
+        Utility.hard_delete_document([CognitionData], bot=current_user.get_bot(), **query, user=current_user.get_user())
+    except DoesNotExist:
+        raise AppException("Some or all records do not exist!")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)